cgmes/cgmes-conversion/src/test/java/com/powsybl/cgmes/conversion/test/export/EquipmentExportTest.java

```diff
@@ -1831,4 +1831,128 @@ private Network prepareNetworkForEQComparison(Network network) {
+                .setId("wind_onshore")
+                .setNode(6)
+                .setMinP(0.0)
+                .setMaxP(100.0)
+                .setTargetP(25.0)
+                .setTargetQ(10.0)
+                .setVoltageRegulatorOn(false)
+                .setEnergySource(EnergySource.WIND)
+                .add();
+        Generator windOffshore = voltageLevel1.newGenerator()
+                .setId("wind_offshore")
+                .setNode(7)
+                .setMinP(0.0)
+                .setMaxP(100.0)
+                .setTargetP(25.0)
+                .setTargetQ(10.0)
+                .setVoltageRegulatorOn(false)
+                .setEnergySource(EnergySource.WIND)
+                .add();
+        topology1.newInternalConnection().setNode1(0).setNode2(1).add();
+        topology1.newInternalConnection().setNode1(0).setNode2(2).add();
+        topology1.newInternalConnection().setNode1(0).setNode2(3).add();
+        topology1.newInternalConnection().setNode1(0).setNode2(4).add();
+        topology1.newInternalConnection().setNode1(0).setNode2(5).add();
+        topology1.newInternalConnection().setNode1(0).setNode2(6).add();
+        topology1.newInternalConnection().setNode1(0).setNode2(7).add();
+        windOnshore.setProperty(Conversion.PROPERTY_WIND_GEN_UNIT_TYPE, "onshore");
+        windOffshore.setProperty(Conversion.PROPERTY_WIND_GEN_UNIT_TYPE, "offshore");
+        return network;
+    }
+
+    @Test
+    void generatingUnitTypesTest() throws IOException {
+        Network network = allGeneratingUnitTypesNetwork();
+
+        // Export as cgmes
+        Path outputPath = tmpDir.resolve("temp.cgmesExport");
+        Files.createDirectories(outputPath);
+        String baseName = "generatingUnitTypes";
+        new CgmesExport().export(network, new Properties(), new DirectoryDataSource(outputPath, baseName));
+
+        // re-import
+        Network actual = new CgmesImport().importData(new DirectoryDataSource(outputPath, baseName), NetworkFactory.findDefault(), new Properties());
+
+        // check the generating unit types
+        assertEquals(EnergySource.OTHER, actual.getGenerator("other").getEnergySource());
+        assertEquals(EnergySource.THERMAL, actual.getGenerator("thermal").getEnergySource());
+        assertEquals(EnergySource.HYDRO, actual.getGenerator("hydro").getEnergySource());
+        assertEquals(EnergySource.NUCLEAR, actual.getGenerator("nuclear").getEnergySource());
+        assertEquals(EnergySource.WIND, actual.getGenerator("wind_offshore").getEnergySource());
+        assertEquals("offshore", actual.getGenerator("wind_offshore").getProperty(Conversion.PROPERTY_WIND_GEN_UNIT_TYPE));
+        assertEquals(EnergySource.WIND, actual.getGenerator("wind_onshore").getEnergySource());
+        assertEquals("onshore", actual.getGenerator("wind_onshore").getProperty(Conversion.PROPERTY_WIND_GEN_UNIT_TYPE));
+        assertEquals(EnergySource.SOLAR, actual.getGenerator("solar").getEnergySource());
```

Comment:
The exported CGMES file shouldn't be reimported back to IIDM. The test should be performed on the CGMES file itself. Otherwise, there is no guarantee that the export itself works.
There is a method in ```ConversionUtil``` that helps searching for a regex in a text file: ```getFirstMatch```. You could use it as follow:
``` java
Pattern windTypePattern = Pattern.compile("<cim:WindGeneratingUnit rdf:ID=\"_wind_onshore_WGU\">.*?" +
                "<cim:WindGeneratingUnit.windGenUnitType rdf:resource=\"http://iec.ch/TC57/2013/CIM-schema-cim16#WindGenUnitKind.(.*?)\"/>", Pattern.DOTALL);
String windType = ConversionUtil.getFirstMatch(exportedFile, windTypePattern);
```

Reply:
There is some guarantee, as long as the mirror unit test exist on importer side. But agree it is better, so I adjusted.
Note that there are many tests currently working with the re-import ... I just copy/pasted an existing one (grep `// re-import` in this test file).